### PR TITLE
[feature](array-type) Unique table support array value

### PR DESCRIPTION
**Pull request by hqx871**
# Proposed changes

Unique table support array value

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by zhannngchen — DISMISSED** [review]
LGTM

**Comment by xiaokang** [comment]
run buildall

File: be/test/vec/aggregate_functions/agg_replace_test.cpp [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+    }
+
+    template <typename DataType, bool nullable>
+    void test_agg_array_replace(const std::string& fn_name, size_t input_nums, size_t expect_num) {
+        vectorized::DataTypePtr data_type = get_data_type<DataType, nullable, true>();
+        DataTypes data_types = {data_type};
+        LOG(INFO) << "test_agg_replace for " << fn_name << "(" << data_types[0]->get_name() << ")";
+        Array array;
+        AggregateFunctionSimpleFactory factory = AggregateFunctionSimpleFactory::instance();
+        auto agg_function = factory.get(fn_name, data_types, array, nullable);
```
[the remark this answers is not in the document]
Comment (amorynan) [reply be/test/vec/aggregate_functions/agg_replace_test.cpp:226]: u can run clang-format

File: be/src/vec/utils/template_helpers.hpp [hunk 2]
```diff
@@ -65,6 +66,7 @@
 
 #define TYPE_TO_COLUMN_TYPE(M)   \
     TYPE_TO_BASIC_COLUMN_TYPE(M) \
+    M(Array, ColumnArray)        \
```
[the remark this answers is not in the document]
Comment (hqx871) [reply be/src/vec/utils/template_helpers.hpp]: hi @xiaokang and @amorynan , I have done to support map and struct type. Please review and rerun the pipeline

**Comment by xiaokang** [comment]
run buildall

**Comment by hqx871** [comment]
run buildall

**Comment by hqx871** [comment]
The regression failed but it seams no business of this pr.

File: be/test/vec/aggregate_functions/agg_replace_test.cpp [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+#include "vec/data_types/data_type_date.h"
+#include "vec/data_types/data_type_date_time.h"
+#include "vec/data_types/data_type_decimal.h"
+#include "vec/data_types/data_type_hll.h"
+#include "vec/data_types/data_type_nullable.h"
+#include "vec/data_types/data_type_number.h"
+#include "vec/data_types/data_type_string.h"
+namespace doris::vectorized {
+
+void register_aggregate_function_replace_reader_load(AggregateFunctionSimpleFactory& factory);
```
[the remark this answers is not in the document]
Comment (amorynan) [reply be/test/vec/aggregate_functions/agg_replace_test.cpp:37]: U can run clang-tidy locally to make c++ code formatted.

**Comment by xiaokang** [comment]
run buildall

**Comment by hqx871** [comment]
run buildall

**Comment by hqx871** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]

**Pull request merged**